### ISSUE TITLE
[Support]: After downgrade to 0.10.1-83481af, Events are not saved anymore

### ISSUE THREAD

**Issue by Larosen** [blakeblackshear/frigate#3273]
### Describe the problem you are having

Regarding the problem I have #3218, wanted to try the new version 0.11.0 Beta 2. But it turned out that this beta could not find my USB TPU. So I uninstalled the beta again and went back to 0.10.1-83481af.
So far so good. TPU is found again and the container runs. Unfortunately, events are no longer saved. 24/7 Recordings still work without problems and can be played. When a person is detected, I get a notification including a picture on my phone through Home Assistant, but no event shows up in Frigate. These are simply no longer saved? Does anyone have an idea how to fix this? Before the futile update to 0.11 everything still worked. I assume that it is due to a changed container variable, which is not easily visible in Home Assistant. 

### Version

0.10.1-83481AF

### Frigate config file

```yaml
mqtt:
  host: 10.20.60.79
  port: 1883
  user: mqtt
  password: ...

detectors:
  coral:
    type: edgetpu
    device: usb


cameras:
  Spielzimmer:
    ffmpeg:
      output_args:
        record: -f segment -segment_time 10 -segment_format mp4 -reset_timestamps 1 -strftime 1 -c copy -tag:v hvc1 -c:a aac
        rtmp: -c copy -f flv -tag:v hvc1
      inputs:
        - path: rtsp://...@10.20.60.87:554/Streaming/Channels/102
          roles:
            - detect  
        - path: rtsp://...@10.20.60.87:554/Streaming/Channels/101
          roles:
            - record
            - rtmp
            #- detect #new
    best_image_timeout: 60
    detect:
      width: 640
      height: 480
      fps: 5
    motion:
      mask:
        - 640,316,640,0,531,0,360,0,350,62
        - 386,336,498,436,508,415,497,346,436,298,414,306
        - 291,0,254,275,116,350,97,37
    mqtt:
      enabled: True
      timestamp: True
      bounding_box: True
      crop: True
      height: 270
      quality: 100



  Sauna:
    ffmpeg:
      inputs:
        - path: rtsp://....@10.20.60.104:556/Streaming/Channels/103
          roles:
            - detect  
        - path: rtsp://....@10.20.60.104:556/Streaming/Channels/101
          roles:
            - record
            - rtmp
    best_image_timeout: 60
    motion:
      mask:
        - 37,105,501,102,506,47,37,42
        - 1920,0,1920,281,1485,313,1046,397,983,0
        - 1072,464,1242,472,1255,341,1082,201
        - 1920,1080,1920,587,1344,514,1189,563,1099,652,837,827,1033,791,1119,945,960,1080
        - 1633,0,1920,0,1920,546,1795,532,1742,551,1698,543,1615,529,1558,446,1563,342
    mqtt:
      enabled: True
      timestamp: True
      bounding_box: True
      crop: True
      height: 270
      quality: 100


  Hintereingang:
    ffmpeg:
      inputs:
        - path: rtsp://...@10.20.60.103:556/Streaming/Channels/103
          roles:
            - detect  
        - path: rtsp://...@10.20.60.103:556/Streaming/Channels/101
          roles:
            - record
            - rtmp
    best_image_timeout: 60
    motion:
      mask:
        - 743,0,1800,0,1789,324,1127,313,550,401,426,571,292,881,0,639,0,469,0,0
    mqtt:
      enabled: True
      timestamp: True
      bounding_box: True
      crop: True
      height: 270
      quality: 100

birdseye:
  enabled: True
  width: 1920
  height: 1080
  quality: 27
  mode: continuous

timestamp_style:
  position: "tl"
  format: "%d/%m/%Y %H:%M:%S"
  color:
    red: 255
    green: 255
    blue: 255
  thickness: 2
  effect: shadow

motion:
  threshold: 50
  contour_area: 30
  delta_alpha: 0.2
  frame_alpha: 0.2
  frame_height: 50
  #mask: 0,900,1080,900,1080,1920,0,1920
  improve_contrast: True


record:
  enabled: True
  expire_interval: 60
  retain:
    days: 3
    mode: all
  events:
    max_seconds: 300
    pre_capture: 10
    post_capture: 15
    retain:
      default: 4
      mode: active_objects
      objects:
        person: 5
        cat: 2
        dog: 1
        car: 1


objects:
  track:
    - person
    - mouse
    - cat
    - dog
    - laptop
    - handbag
    - umbrella
    - backpack
    - cell phone
    - bycicle
    - cow
    - sheep
    - bird
    - bear
  #mask: 0,0,1000,0,1000,200,0,200
  filters:
    person:
      min_area: 1000
      max_area: 400000
      min_score: 0.6
      threshold: 0.75
    laptop:
      min_score: 0.8
      threshold: 0.9
    bird:
      min_score: 0.8
      threshold: 0.9
      max_area: 90000
    dog:
      min_score: 0.7
      threshold: 0.8
    umbrella:
      min_score: 0.7
      threshold: 0.84
      max_area: 202000
    mouse: 
      min_score: 0.7
      threshold: 0.84
      max_area: 6000

ffmpeg:
  global_args: -hide_banner -loglevel warning
  hwaccel_args: []
  input_args: -avoid_negative_ts make_zero -fflags +genpts+discardcorrupt -rtsp_transport tcp -stimeout 5000000 -use_wallclock_as_timestamps 1
  output_args:
    detect: -f rawvideo -pix_fmt yuv420p
    record: -f segment -segment_time 10 -segment_format mp4 -reset_timestamps 1 -strftime 1 -c:v copy -c:a aac
    rtmp: -c:v copy -c:a aac -f flv

detect:
  width: 1920
  height: 1080
  fps: 5
  enabled: True
  max_disappeared: 25
  stationary:
    interval: 10
    max_frames:
      objects:
        person: 1000
        cat: 5000

rtmp:
  enabled: False

snapshots:
  enabled: True
  timestamp: True
  bounding_box: True
  crop: True
  #height: 175
  required_zones: []
  retain:
    default: 10
    objects:
      person: 15
      cat: 10
      bear: 5
      dog: 5
```

### Relevant log output

```shell
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[2022-05-29 11:53:53] frigate.app                    INFO    : Starting Frigate (0.10.1-83481af)
[2022-05-29 11:53:53] frigate.app                    INFO    : Creating directory: /tmp/cache
Starting migrations
[2022-05-29 11:53:53] peewee_migrate                 INFO    : Starting migrations
There is nothing to migrate
[2022-05-29 11:53:53] peewee_migrate                 INFO    : There is nothing to migrate
[2022-05-29 11:53:53] frigate.app                    INFO    : Output process started: 216
[2022-05-29 11:53:53] detector.coral                 INFO    : Starting detection process: 215
[2022-05-29 11:53:53] frigate.app                    INFO    : Camera processor started for Spielzimmer: 219
[2022-05-29 11:53:53] ws4py                          INFO    : Using epoll
[2022-05-29 11:53:53] frigate.edgetpu                INFO    : Attempting to load TPU as usb
[2022-05-29 11:53:53] frigate.app                    INFO    : Camera processor started for Sauna: 222
[2022-05-29 11:53:53] frigate.app                    INFO    : Camera processor started for Hintereingang: 224
[2022-05-29 11:53:53] frigate.app                    INFO    : Capture process started for Spielzimmer: 229
[2022-05-29 11:53:53] frigate.app                    INFO    : Capture process started for Sauna: 230
[2022-05-29 11:53:53] frigate.app                    INFO    : Capture process started for Hintereingang: 233
[2022-05-29 11:53:53] ws4py                          INFO    : Using epoll
[2022-05-29 11:53:56] frigate.edgetpu                INFO    : TPU found
[2022-05-29 11:54:43] frigate.watchdog               INFO    : Detection appears to be stuck. Restarting detection process...
[2022-05-29 11:54:43] root                           INFO    : Waiting for detection process to exit gracefully...
[2022-05-29 11:55:13] root                           INFO    : Detection process didnt exit. Force killing...
[2022-05-29 11:55:13] detector.coral                 INFO    : Starting detection process: 351
[2022-05-29 11:55:13] frigate.edgetpu                INFO    : Attempting to load TPU as usb
[2022-05-29 11:55:16] frigate.edgetpu                INFO    : TPU found
[2022-05-29 11:58:33] frigate.watchdog               INFO    : Detection appears to be stuck. Restarting detection process...
[2022-05-29 11:58:33] root                           INFO    : Waiting for detection process to exit gracefully...
[2022-05-29 11:59:03] root                           INFO    : Detection process didnt exit. Force killing...
[2022-05-29 11:59:03] detector.coral                 INFO    : Starting detection process: 476
[2022-05-29 11:59:03] frigate.edgetpu                INFO    : Attempting to load TPU as usb
[2022-05-29 11:59:06] frigate.edgetpu                INFO    : TPU found
[2022-05-29 12:01:33] frigate.watchdog               INFO    : Detection appears to be stuck. Restarting detection process...
[2022-05-29 12:01:33] root                           INFO    : Waiting for detection process to exit gracefully...
[2022-05-29 12:02:04] root                           INFO    : Detection process didnt exit. Force killing...
[2022-05-29 12:02:04] detector.coral                 INFO    : Starting detection process: 578
[2022-05-29 12:02:04] frigate.edgetpu                INFO    : Attempting to load TPU as usb
[2022-05-29 12:02:06] frigate.edgetpu                INFO    : TPU found
[2022-05-29 12:05:54] frigate.watchdog               INFO    : Detection appears to be stuck. Restarting detection process...
[2022-05-29 12:05:54] root                           INFO    : Waiting for detection process to exit gracefully...
[2022-05-29 12:06:24] root                           INFO    : Detection process didnt exit. Force killing...
[2022-05-29 12:06:24] detector.coral                 INFO    : Starting detection process: 718
[2022-05-29 12:06:24] frigate.edgetpu                INFO    : Attempting to load TPU as usb
[2022-05-29 12:06:27] frigate.edgetpu                INFO    : TPU found
[2022-05-29 12:09:54] frigate.watchdog               INFO    : Detection appears to be stuck. Restarting detection process...
[2022-05-29 12:09:54] root                           INFO    : Waiting for detection process to exit gracefully...
[2022-05-29 12:10:15] ws4py                          INFO    : Managing websocket [Local => 127.0.0.1:5002 | Remote => 127.0.0.1:45494]
[2022-05-29 12:10:24] root                           INFO    : Detection process didnt exit. Force killing...
[2022-05-29 12:10:24] detector.coral                 INFO    : Starting detection process: 856
[2022-05-29 12:10:24] frigate.edgetpu                INFO    : Attempting to load TPU as usb
[2022-05-29 12:10:27] frigate.edgetpu                INFO    : TPU found
[2022-05-29 12:10:46] ws4py                          INFO    : Terminating websocket [Local => 127.0.0.1:5002 | Remote => 127.0.0.1:45494]
[2022-05-29 12:11:57] ws4py                          INFO    : Managing websocket [Local => 127.0.0.1:5002 | Remote => 127.0.0.1:43290]
[2022-05-29 12:12:12] ws4py                          INFO    : Managing websocket [Local => 127.0.0.1:8082 | Remote => 127.0.0.1:38752]
Exception in thread event_processor:
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/peewee.py", line 3129, in execute_sql
    cursor.execute(sql, params or ())
sqlite3.IntegrityError: NOT NULL constraint failed: event.retain_indefinitely
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
    self.run()
  File "/opt/frigate/frigate/events.py", line 67, in run
    Event.replace(
  File "/usr/local/lib/python3.8/dist-packages/peewee.py", line 1898, in inner
    return method(self, database, *args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/peewee.py", line 1969, in execute
    return self._execute(database)
  File "/usr/local/lib/python3.8/dist-packages/peewee.py", line 2730, in _execute
    return super(Insert, self)._execute(database)
  File "/usr/local/lib/python3.8/dist-packages/peewee.py", line 2466, in _execute
    return self.handle_result(database, cursor)
  File "/usr/local/lib/python3.8/dist-packages/peewee.py", line 2739, in handle_result
    return database.last_insert_id(cursor, self._query_type)
  File "/usr/local/lib/python3.8/dist-packages/peewee.py", line 3218, in last_insert_id
    return cursor.lastrowid
  File "/usr/local/lib/python3.8/dist-packages/playhouse/sqliteq.py", line 88, in lastrowid
    self._wait()
  File "/usr/local/lib/python3.8/dist-packages/playhouse/sqliteq.py", line 63, in _wait
    raise self._exc
  File "/usr/local/lib/python3.8/dist-packages/playhouse/sqliteq.py", line 178, in execute
    cursor = self.database._execute(obj.sql, obj.params, obj.commit)
  File "/usr/local/lib/python3.8/dist-packages/peewee.py", line 3136, in execute_sql
    self.commit()
  File "/usr/local/lib/python3.8/dist-packages/peewee.py", line 2902, in __exit__
    reraise(new_type, new_type(exc_value, *exc_args), traceback)
  File "/usr/local/lib/python3.8/dist-packages/peewee.py", line 185, in reraise
    raise value.with_traceback(tb)
  File "/usr/local/lib/python3.8/dist-packages/peewee.py", line 3129, in execute_sql
    cursor.execute(sql, params or ())
peewee.IntegrityError: NOT NULL constraint failed: event.retain_indefinitely
```

### FFprobe output from your camera

```shell
Not necessary
```

### Frigate stats

```json
{"Hintereingang":{"camera_fps":5.1,"capture_pid":233,"detection_fps":0.1,"pid":224,"process_fps":0.2,"skipped_fps":0.0},"Sauna":{"camera_fps":5.1,"capture_pid":230,"detection_fps":0.0,"pid":222,"process_fps":5.1,"skipped_fps":0.0},"Spielzimmer":{"camera_fps":5.0,"capture_pid":229,"detection_fps":0.0,"pid":219,"process_fps":5.0,"skipped_fps":0.0},"detection_fps":0.1,"detectors":{"coral":{"detection_start":0.0,"inference_speed":35.25,"pid":1071}},"service":{"storage":{"/dev/shm":{"free":3085.4,"mount_type":"tmpfs","total":3110.1,"used":24.8},"/media/frigate/clips":{"free":1122523.4,"mount_type":"ext4","total":1373204.4,"used":198082.2},"/media/frigate/recordings":{"free":1122523.4,"mount_type":"ext4","total":1373204.4,"used":198082.2},"/tmp/cache":{"free":3095.1,"mount_type":"tmpfs","total":3110.1,"used":15.1}},"temperatures":{},"uptime":1352,"version":"0.10.1-83481af"}}
```

### Operating system

HassOS

### Install method

HassOS Addon

### Coral version

USB

### Network connection

Wired

### Camera make and model

8MP 4k PoE Cameras

### Any other information that may be helpful

Synology 920+, VM with Home Assistant and Frigate as an Add-On. USB TPU is in a USB Hub with external power.

**Comment by NickM-27** [comment]
You can not switch between versions at will, 0.10 -> 0.11 there are a lot of database migrations and new fields. If you go back to 0.10 the database is now incompatible 

You need to restore the db from a backup or delete it to start fresh.

**Comment by Larosen** [comment]
> You can not switch between versions at will, 0.10 -> 0.11 there are a lot of database migrations and new fields. If you go back to 0.10 the database is now incompatible 
> 
> You need to restore the db from a backup or delete it to start fresh. 

That's what I was afraid of. But how do you delete the database in HomeAssistant? It doesn't work via the frontend, does it?

**Comment by NickM-27** [comment]
> That's what I was afraid of. But how do you delete the database in HomeAssistant? It doesn't work via the frontend, does it?

If you open up the /media/frigate folder you will see frigate.db and potentially a couple other frigate-db... files, delete those and should be good to go.

**Comment by Larosen** [comment]
Cleaning up /media/frigate via SAMBA worked smoothly. I then tried 0.11.0 Beta 2 again, but the frigate screen remains blank, also otherwise this version does not seem to start properly. I will try again with Beta 3, because the Log wasn´t showing any helpful information that aren´t posed in other issues already. 

After deleting the database again, I then reinstalled 0.10.1 and now it finally works as intended. Thank you very much @NickM-27!
